Math/src/main/java/org/tribuo/math/distance/DistanceType.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.tribuo.math.distance;
+
+import org.tribuo.math.la.SGDVector;
+
+public enum DistanceType {
```

Comment:
This is currently duplicated in HdbscanTrainer, KMeansTrainer and KNNTrainer. I appreciate that this can't be refactored out of these classes in a point release, but I'm proposing this for consideration in a future major release.

Reply:
A mapping method seems like a good temporary approach. Once these initial changes get closer the integration with Hdbscan or KMeans can be proposed. The javadoc is added.